src/arch/ssse3.rs

```diff
@@ -0,0 +1,261 @@
+#[cfg(target_arch = "x86")]
+use std::arch::x86::*;
+#[cfg(target_arch = "x86_64")]
+use std::arch::x86_64::*;
+
+#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+#[target_feature(enable = "ssse3")]
```

Comment:
Read up on this and semantics might be different than what is implied by the usage here. This instructs `rustc` to use ssse3 for _its own code gen_ of this function. This is important for llvm's auto-vectorizer and other intrinsics. However, when you're using intrinsics then you're instructing it to place those exact primitive regardless. Thus I think this may be superfluous. It doesn't hurt anyways.

Reply:
So I don't *really* know what is going on here, but disabling target_feature seems to **significantly** slow things down, so I'd rather leave it here :D

Reply:
Interesting. I'd really like to find out why that is but indeed, then let's leave it here, and that a task for myself :joy: